android-agent/src/main/java/io/opentelemetry/android/internal/features/persistence/DiskManager.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.internal.features.persistence
+
+import android.util.Log
+import io.opentelemetry.android.common.RumConstants
+import io.opentelemetry.android.features.diskbuffering.DiskBufferingConfiguration
+import io.opentelemetry.android.internal.services.CacheStorageService
+import io.opentelemetry.android.internal.services.PreferencesService
+import java.io.File
+import java.io.IOException
+
+internal class DiskManager(
+    private val cacheStorageService: CacheStorageService,
+    private val preferencesService: PreferencesService,
+    private val diskBufferingConfiguration: DiskBufferingConfiguration,
+) {
+    @get:Throws(IOException::class)
+    val signalsBufferDir: File
+        get() {
```

Comment:
I _think_ this is fairly idiomatic these days....but I'm happy to be talked out of this style/syntax if it is important to anyone.

Reply:
That's fair enough, considering that Kotlin allows you to access getter methods as if they were class fields later on, it seems like the question of defining them like this vs regular getters boils down to personal preferences really. I've seen custom accessors widely used for cases where the `get()` or `set()` body is quite small, like this one, although it does seem a bit strange to me to see them as big as the `get()` in `maxFolderSize` for example.

Regarding my personal preferences, I find it easier to read and quickly set apart in my mind what parts in a class hold business logic vs what others only store things in memory by using regular methods and fields, although it's probably a matter of getting used to different styles, so it's not the end of the world for me to use either approach.

Reply:
That's fair enough, considering that Kotlin allows you to access getter methods as if they were class fields later on, it seems like the question of defining them like this vs regular getters boils down to personal preferences really. I've seen custom accessors widely used for cases where the `get()` or `set()` body is quite small, like this one, although it does seem a bit strange to me to see them as big as the `get()` in `maxFolderSize` for example.

Regarding my personal preferences, I find it easier to read and quickly set apart in my mind what parts in a class hold business logic vs what others only store things in memory by using regular methods and fields, although it's probably a matter of getting used to different styles, so it's not the end of the world for me to use either approach.